component/src/main/java/org/wso2/carbon/identity/authenticator/nuxeo/NuxeoAuthenticator.java

```diff
@@ -0,0 +1,404 @@
+        }
+        return accessRequest;
+    }
+
+    /**
+     * Get the user info from user info endpoint.
+     *
+     * @param url         User info endpoint.
+     * @param accessToken Access token.
+     * @return Response string.
+     */
+    private String getUserInfoJson(String url, String accessToken) throws AuthenticationFailedException {
+
+        if (log.isDebugEnabled()) {
+            log.debug("User info endpoint: " + url);
+        }
+        BufferedReader reader = null;
+        StringBuilder result = null;
+        Integer statusCode;
+        String line;
+        HttpGet httpget = null;
+        try (CloseableHttpClient httpclient = HttpClients.createDefault()) {
+            httpget = new HttpGet(url);
+            httpget.setHeader(NuxeoAuthenticatorConstants.AUTHORIZATION,
+                    NuxeoAuthenticatorConstants.AUTHENTICATION_BEARER + accessToken);
+            result = new StringBuilder();
+            statusCode=httpclient.execute(httpget).getStatusLine().getStatusCode();
+            reader = new BufferedReader(new InputStreamReader(httpclient.execute(httpget).getEntity().getContent()));
+            while ((line = reader.readLine()) != null) {
+                result.append(line);
+            }
+            if (statusCode == HttpStatus.SC_OK) {
+                if (log.isDebugEnabled()) {
+                    log.debug("logged in User info: " + result.toString());
+                }
+                return result.toString();
+            } else {
+                if (log.isDebugEnabled()) {
+                    log.debug("Couldn't get the successful response while getting the " +
+                            "UserInfo for the logged in User. Status code: " + statusCode + " and Response: " +
+                            result.toString());
+                }
+                throw new AuthenticationFailedException(" Couldn't get the successful response while getting the " +
+                        "UserInfo for the logged in User. Status code: " + statusCode + " and Response: " +
+                        result.toString());
+            }
+        } catch (ClientProtocolException e) {
+            throw new AuthenticationFailedException("Exception occurred from the Malformed url while " +
+                    "getting the user Information. Malformed url: " + httpget.getURI(), e);
+        } catch (IOException e) {
+            throw new AuthenticationFailedException("Exception occurred from the connection was aborted. " +
+                    "(eg. invalid hostname, no server listening or internet connection was lost) " +
+                    "while getting the UserInfo. ", e);
+        }
+    }
+
+    /**
+     * Get the user claims from user info.
+     *
+     * @param userInfo User info.
+     * @return Response Map<ClaimMapping, String>.
+     */
+    protected Map<ClaimMapping, String> mapUserInfo(String userInfo) {
+
+        if (userInfo.isEmpty()) {
```

Comment:
What's the purpose of this if block ? I guess there's no further processing required if the userinfo is empty.

Reply:
Also, without checking empty here, we can do this check before calling `mapUserInfo` and throw an error